File: libraryCoreJava/src/main/java/com/themone/core/base/impl/BaseActivity.java
package com.themone.core.base.impl;

import android.app.Activity;
import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import com.themone.core.util.StatusBarUtil;

/**
 * @author zhiqiang
 * @date 2019-06-04
 * @desc 简单 Activity，无业务逻辑可继承此类
 */
public class BaseActivity extends AppCompatActivity {

    protected Activity mContext;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        StatusBarUtil.setTransparentForWindow(this);
        mContext = this;
    }

    @Override
    public void setContentView(int layoutResID) {
        super.setContentView(layoutResID);
        if (isFitsSystemWindows()) {
            StatusBarUtil.setFitsSystemWindows(this);
        }
    }

    @Override
    public void onLowMemory() {
        super.onLowMemory();
    }

    public boolean isFitsSystemWindows() {
        return true;
    }
}

File: libraryFramework/src/main/java/com/theone/framework/base/BaseActivity.kt
package com.theone.framework.base

import android.content.Context
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import com.themone.core.base.impl.CoreActivity

import com.theone.framework.util.I18NUtil
import com.theone.framework.util.StatusBarUtil
import io.reactivex.rxjava3.disposables.CompositeDisposable

/**
 * @Author zhiqiang
 * @Date 2019-06-19
 * @Email liuzhiqiang@theone.com
 * @Description
 */
open class BaseActivity : CoreActivity(), IBaseActivity {
    override val compositeDisposable: CompositeDisposable = CompositeDisposable()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //状态栏透明
        StatusBarUtil.transparentStatusBar(this)
    }

    override fun setContentView(layoutResID: Int) {
        super.setContentView(layoutResID)
        //状态栏偏移
        if (isFitsSystemWindows()) {
            StatusBarUtil.setFitsSystemWindows(this)
        }
    }

    override fun setContentView(view: View?) {
        super.setContentView(view)
        //状态栏偏移
        if (isFitsSystemWindows()) {
            StatusBarUtil.setFitsSystemWindows(this)
        }
    }

    override fun setContentView(view: View?, params: ViewGroup.LayoutParams?) {
        super.setContentView(view, params)
        //状态栏偏移
        if (isFitsSystemWindows()) {
            StatusBarUtil.setFitsSystemWindows(this)
        }
    }

    /**
     * 1.statusBar的底色来自Imageview等元素，而非根部局
     * 2.statusBar的底色Fragment
     * 以上情况，isFitsSystemWindows()设置为false，且手动调用[StatusBarUtil.addStatusBarOffsetForView]
     * @return Boolean
     */
    open fun isFitsSystemWindows(): Boolean {
        return false
    }

    override fun attachBaseContext(newBase: Context) {
        super.attachBaseContext(I18NUtil.updateResource(newBase))
    }

    override fun onDestroy() {
        compositeDisposable.clear()
        super.onDestroy()
    }
}